lib/datasets.go

```diff
@@ -982,10 +982,13 @@ func (datasetImpl) Save(scope scope, p *SaveParams) (*dataset.Dataset, error) {
 
 		scriptOut := p.ScriptOutput
 		secrets := p.Secrets
-		// allocate an ID for the transform, subscribe to print output & build up
-		// runState
-		runID := run.NewID()
-		runState = run.NewState(runID)
+		runID := ds.Commit.RunID
+		if runID == "" {
+			// if there is no given runID, allocate an ID for the transform,
+			// subscribe to print output & build up the run.State
+			runID = run.NewID()
+		}
+		runState = &run.State{ID: runID}
```

Comment:
Okay, big question here. In the spike automation pr, we pass in a runID in the `dataset.Commit.RunID` field. Wondering if instead we should expand the `SaveParams` to have a `RunID` field, and pass it through that way.

Reply:
Dope, want to get @b5 & @Arqu 's eyes on this before I make the change cause these lib params feel settled since our api refactoring and this would technically be a change to the api as well.

Reply:
so at @b5 request, we are leaving as is!